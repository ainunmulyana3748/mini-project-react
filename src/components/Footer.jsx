import React from "react";

const Footer = () => {
  return (
    <footer className="bg-black text-white py-8 mt-12">
      <div className="max-w-screen-xl mx-auto px-4 grid grid-cols-1 md:grid-cols-3 gap-6 text-sm text-center md:text-left place-items-center md:place-items-start">
        {/* Brand */}
        <div>
          <h2 className="text-xl font-bold mb-2">Ditemenin.</h2>
          <p className="text-gray-400 max-w-xs">
            Temukan tour guide profesional untuk menemanimu di berbagai
            destinasi seru.
          </p>
        </div>

        {/* Download App */}
        <div>
          <h3 className="font-semibold mb-2">Download Ditemenin App</h3>
          <div className="flex items-center md:items-start gap-4">
            <a href="#" target="_blank" rel="noopener noreferrer">
              <img
                src="https://d1785e74lyxkqq.cloudfront.net/_next/static/v3/f/f519939e72eccefffb6998f1397901b7.svg"
                alt="Get it on Google Play"
                className="h-10"
              />
            </a>
            <a href="#" target="_blank" rel="noopener noreferrer">
              <img
                src="https://d1785e74lyxkqq.cloudfront.net/_next/static/v3/1/18339f1ae28fb0c49075916d11b98829.svg"
                alt="Download on the App Store"
                className="h-10"
              />
            </a>
          </div>
        </div>

        {/* Kontak / Sosial */}
        <div>
          <h3 className="font-semibold mb-2">Kontak</h3>
          <p className="text-gray-400">Email: support@ditemenin.com</p>
          <p className="text-gray-400">Instagram: @ditemenin</p>
        </div>
      </div>

      <div className="mt-8 text-center text-gray-500 text-xs">
        &copy; {new Date().getFullYear()} Ditemenin. All rights reserved.
      </div>
    </footer>
  );
};

export default Footer;
