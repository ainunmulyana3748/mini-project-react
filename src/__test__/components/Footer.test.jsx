import { render, screen } from "@testing-library/react";
import Footer from "../../components/Footer";

describe("Footer Component", () => {
  test("render brand, download app, dan kontak", () => {
    render(<Footer />);
    expect(screen.getByText("Ditemenin.")).toBeInTheDocument();
    expect(screen.getByText("Download Ditemenin App")).toBeInTheDocument();
    expect(screen.getByAltText("Get it on Google Play")).toBeInTheDocument();
    expect(
      screen.getByAltText("Download on the App Store")
    ).toBeInTheDocument();
    expect(screen.getByText("Kontak")).toBeInTheDocument();
    expect(
      screen.getByText("Email: support@ditemenin.com")
    ).toBeInTheDocument();
    expect(screen.getByText("Instagram: @ditemenin")).toBeInTheDocument();
  });

  test("copyright tahun sesuai tahun sekarang", () => {
    render(<Footer />);
    const year = new Date().getFullYear();
    expect(
      screen.getByText(new RegExp(`${year} Ditemenin. All rights reserved.`))
    ).toBeInTheDocument();
  });

  test("snapshot footer", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
