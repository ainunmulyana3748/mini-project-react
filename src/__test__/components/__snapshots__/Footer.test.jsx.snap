// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component snapshot footer 1`] = `
<div>
  <footer
    class="bg-black text-white py-8 mt-12"
  >
    <div
      class="max-w-screen-xl mx-auto px-4 grid grid-cols-1 md:grid-cols-3 gap-6 text-sm text-center md:text-left place-items-center md:place-items-start"
    >
      <div>
        <h2
          class="text-xl font-bold mb-2"
        >
          Ditemenin.
        </h2>
        <p
          class="text-gray-400 max-w-xs"
        >
          Temukan tour guide profesional untuk menemanimu di berbagai destinasi seru.
        </p>
      </div>
      <div>
        <h3
          class="font-semibold mb-2"
        >
          Download Ditemenin App
        </h3>
        <div
          class="flex items-center md:items-start gap-4"
        >
          <a
            href="#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Get it on Google Play"
              class="h-10"
              src="https://d1785e74lyxkqq.cloudfront.net/_next/static/v3/f/f519939e72eccefffb6998f1397901b7.svg"
            />
          </a>
          <a
            href="#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Download on the App Store"
              class="h-10"
              src="https://d1785e74lyxkqq.cloudfront.net/_next/static/v3/1/18339f1ae28fb0c49075916d11b98829.svg"
            />
          </a>
        </div>
      </div>
      <div>
        <h3
          class="font-semibold mb-2"
        >
          Kontak
        </h3>
        <p
          class="text-gray-400"
        >
          Email: support@ditemenin.com
        </p>
        <p
          class="text-gray-400"
        >
          Instagram: @ditemenin
        </p>
      </div>
    </div>
    <div
      class="mt-8 text-center text-gray-500 text-xs"
    >
      Â© 
      2025
       Ditemenin. All rights reserved.
    </div>
  </footer>
</div>
`;
